rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6229,6 +6228,292 @@ void receiveHeader_block_exist_in_storage() throws IOException, BlockStoreExcept
+                bridgeMainnetConstants,
+                activations
+            );
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainnetConstants)
+                .withProvider(bridgeStorageProvider)
+                .withRepository(repository)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withActivations(activations)
+                .build();
+            btcBlockStoreWithCache = btcBlockStoreFactory.newInstance(repository, bridgeMainnetConstants, bridgeStorageProvider, activations);
+
+            // Create block 849134 with cumulative difficulty
+            BigInteger block849134ChainWork = new BigInteger("00000000000000000000000000000000000000007ffef81fa11393037c9df17b", 16);
+            StoredBlock block849134ToStore = new StoredBlock(block849134, block849134ChainWork, 849134);
+
+            // save block 849134 in storage
+            btcBlockStoreWithCache.put(block849134ToStore);
+            btcBlockStoreWithCache.setChainHead(block849134ToStore);
+            repository.save();
+            // assert that block 849134 was correctly saved
+            assertEquals(btcBlockStoreWithCache.getChainHead().getHeader().getHash(), block849134.getHash());
+
+            BtcBlock[] headersToSend = new BtcBlock[] { block849135, block849136, block849137, blockWithTooMuchWork, block849139 };
+            // assert blocks until 849137 are correctly saved
+            // and blocks from 849138 are not saved
+            bridgeSupport.receiveHeaders(headersToSend);
+            assertNotNull(btcBlockStoreWithCache.get(block849135.getHash()));
+            assertNotNull(btcBlockStoreWithCache.get(block849136.getHash()));
+            assertNotNull(btcBlockStoreWithCache.get(block849137.getHash()));
+            assertNull(btcBlockStoreWithCache.get(blockWithTooMuchWork.getHash()));
+            assertNull(btcBlockStoreWithCache.get(block849139.getHash()));
+        }
+
+        @Test
+        void receiveHeader_afterRSKIP434_returnsSuccessfulAndSavesTheBlock() throws BlockStoreException, IOException {
+            // rskip434 should be active
+            activations = ActivationConfigsForTest.all().forBlock(0);
+            bridgeStorageProvider = new BridgeStorageProvider(
+                repository,
+                PrecompiledContracts.BRIDGE_ADDR,
+                bridgeMainnetConstants,
+                activations
+            );
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainnetConstants)
+                .withProvider(bridgeStorageProvider)
+                .withRepository(repository)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withActivations(activations)
+                .build();
+            btcBlockStoreWithCache = btcBlockStoreFactory.newInstance(repository, bridgeMainnetConstants, bridgeStorageProvider, activations);
```

Comment:
This setup is repeated three times. Is it possible to move it outside the test and then reuse it where's needed?

Reply:
we need different activations for each test, so we have to recreate the storage provider and bridge support each time.
What i can do is removing the `activations` from the `setUp`, thanks!!